File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let answer = [];
    const days = progresses.map((p,i)=>{
        const remain = 100 - p
        const speed = speeds[i];
        if (remain % speed !== 0){
            return Math.floor(remain/speed)+1; 
        }
        return Math.floor(remain/speed);
    })
    const len = days.length;
    let stack = [];
    for (let i = 0; i < len; i++){
        if (stack.length > 0){
            if (stack[0] >= days[i]){
                stack.push(days[i]);
            } else {
                answer.push(stack.length);
                stack = [days[i]];
            }
        } else {
            stack.push(days[i]);
        }
    }
    if (stack.length > 0){
        answer.push(stack.length);
    }
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    days = []
    for p,s in zip(progresses,speeds):
        day = (100-p)//s
        if (100-p)%s>0:
            day+=1
        days.append(day)
    
    m = 0
    for d in days:
        if (d > m):
            answer.append(1)
            m = d
        else:
            answer[-1] += 1
            
    return answer